pkg/nodes/list.go

```diff
@@ -60,3 +68,110 @@ func ListExternalIPv4Networks(apiClient *clients.Settings, options v1.ListOption
 
 	return ipV4ExternalAddresses, nil
 }
+
+// WaitForAllNodesAreReady waits for all nodes to be Ready for a time duration up to the timeout.
+func WaitForAllNodesAreReady(apiClient *clients.Settings,
+	timeout time.Duration,
+	options v1.ListOptions) (bool, error) {
+	glog.V(100).Infof("Waiting for all nodes to be in the Ready state for up to a duration of %v",
+		timeout)
+
+	nodesList, err := List(apiClient, options)
+	if err != nil {
+		glog.V(100).Infof("Failed to list all nodes due to %s", err.Error())
+
+		return false, err
+	}
+
+	// Wait 5 secs in each iteration before condition function () returns true or errors or times out
+	// after availableDuration
+	err = wait.PollImmediate(backoff, timeout, func() (done bool, err error) {
+		// iterate through the ClusterOperators in the list.
+		for _, node := range nodesList {
+			_, err = node.IsReady()
```

Comment:
We need to update this to check the boolean returned from the IsReady method
```suggestion
			ready, err := node.IsReady()
			if err != nil {
				glog.V(100).Infof("Node %s has error %w", node.Object.Name, err)

				return false, err
			}
			
			if !ready {
			    glog.V(100).Infof("Node %s not Ready", node.Object.Name)
			    
		            return false, nil
		       }
```